components/butterfly/src/rumor/heat.rs

```diff
@@ -125,6 +125,32 @@ impl RumorHeat {
             }
         }
     }
+
+    /// When a member is considered "gone" (e.g., once it is
+    /// considered Departed), we can get rid of all the "cooling"
+    /// information, since we're not going to be sending anything
+    /// their way again.
+    ///
+    /// Without this, we would continue carrying around this
+    /// information for Supervisors that we're never going to see
+    /// again. The larger the network of Supervisors is, the more
+    /// memory this consumes. If that member should ever come back
+    /// again, all rumors would be considered "hot" for them, so they
+    /// will get a bit more network traffic initially.
+    pub fn purge(&self, id: &str) {
+        let mut heat_map = self.0.write().expect("RumorHeat lock poisoned");
+
+        // Remove any information about Service rumors for this
+        // particular member... it's leaving, so none of its services
+        // will be around either.
+        heat_map.retain(|k, _| !(k.kind == RumorType::Service && k.id == id));
+
+        // Remove any "cooling" information for this member, across
+        // all types of rumors.
+        for heat in heat_map.values_mut() {
+            heat.remove(id);
+        }
```

Comment:
Perhaps either quantifying or summarizing the bytes purged might be of some use here?

```
+use std::mem;
...
         for heat in heat_map.values_mut() {
-            heat.remove(id);
+            if let Some(r) = heat.remove(id) {
+                debug!("Purging rumor heat of size {} bytes", mem::size_of_val(&r));
+            } else {
+                debug!("Member {} unexpectedly had no rumor heat to purge.", id);
+            }
         }
```

Reply:
`size_of_val` doesn't always give you what you think you want, since it's the size of the _value_, which may not be the same as the "total size" of the value, including any heap allocations. For example, you'll always get `24` if your value is a `String`, since a `String` is a 3-byte "fat pointer", regardless of how many characters are in the actual character string pointed to on the heap.

However, in this case, it's always going to be 64 bytes; 56 for the member ID (24 for the "fat pointer" and 32 for the ID string itself) and 8 for the `usize` counter.